WORKSPACE

```diff
@@ -2,19 +2,14 @@ workspace(name = "envoy_openssl")
 
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-http_archive(
-      name = "openssl",
-      urls = ["https://github.com/openssl/openssl/archive/OpenSSL_1_1_1d.tar.gz"],
-      sha256 = "23011a5cc78e53d0dc98dfa608c51e72bcd350aa57df74c5d5574ba4ffb62e74",
-      build_file = "@//:openssl.BUILD",
-      strip_prefix = "openssl-OpenSSL_1_1_1d",
-)
+load("//:openssl.bzl", "openssl_repository", "openssl_shared_repository")
 
-new_local_repository(
-    name = "openssl_shared",
-    path = "/usr/lib/x86_64-linux-gnu",
-    build_file = "openssl_host_shared.BUILD"
-)
+# If you need OpenSSL dynamically linked to Envoy then comment the line
```

Comment:
just a thought but could we make this more user friendly by providing a `config_setting` for easy user selection. I'm thinking if it would work the same as `boringssl` vs `boringssl_fips`?

Reply:
Can `config_setting` be used in WORKSPACE files? I'm looking at Bazel documentation and I see it only being used in BUILD files. Which means that I would need to use `select` inside `deps` attributes in all BUILD files, like:

```
config_setting(
    name = "openssl_shared",
    values = {"define": "openssl=shared"},
)

envoy_cc_library(
    name = "bssl_compat_lib",
    repository = "@envoy",
    srcs = [
        "bssl.cc",
    ],
    hdrs = [
        "bssl.h",
    ],
    deps = select({
        ":openssl_shared": ["@openssl_shared//:ssl"],
        "//conditions:default": ["@openssl//:ssl"],
    }),
)
```

And I can't imagine how injecting openssl into Envoy would look like with that solution.

Reply:
I think `config_setting` would work as a solution if we want to submit OpenSSL specific settings to upstream Envoy repository, am I wrong?

Reply:
@mrostecki (sorry for the delay). I'm not a bazel expert but my idea was we could try to follow Envoy's example with boringssl and have the selection in this repository.

btw, we're in slack (#envoy-openssl) if you wish to brainstorm this there